synapseclient/models/protocols/json_schema_protocol.py

```diff
@@ -0,0 +1,135 @@
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaBinding: An object containing details about the JSON schema binding.
+        """
+        return JSONSchemaBinding()
+
+    def get_schema(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> "JSONSchemaBinding":
+        """
+        Get the JSON schema bound to the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaBinding: An object containing details about the bound JSON schema.
+        """
+        return JSONSchemaBinding()
+
+    def delete_schema(self, *, synapse_client: Optional["Synapse"] = None) -> None:
+        """
+        Delete the JSON schema bound to the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+        """
+
+    def validate_schema(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> Union["JSONSchemaValidation", "InvalidJSONSchemaValidation"]:
+        """
+        Validate the entity against the bound JSON schema.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            Union[JSONSchemaValidation, InvalidJSONSchemaValidation]: The validation results.
+        """
+        return InvalidJSONSchemaValidation() or JSONSchemaValidation()
+
+    def get_schema_derived_keys(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> "JSONSchemaDerivedKeys":
+        """
+        Retrieve derived JSON schema keys for the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaDerivedKeys: An object containing the derived keys for the entity.
+        """
+        return JSONSchemaDerivedKeys()
+
+
+class ContainerEntityJSONSchemaProtocol(BaseJSONSchemaProtocol):
+    """
+    Mixin class to provide JSON schema functionality for container entities.
+    This class extends BaseJSONSchemaProtocol and is intended for use with Synapse container entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the container entity.
+    """
+
+    async def get_schema_validation_statistics_async(
```

Comment:
The methods in this class will need to be updated to remove the async keywords

Reply:
Got it. Thanks for catching it.